ts/keys.go

```diff
@@ -0,0 +1,114 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Matt Tracy (matt.r.tracy@gmail.com)
+
+package ts
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/storage/engine"
+	"github.com/cockroachdb/cockroach/util/encoding"
+)
+
+// Time series keys are carefully constructed to usefully sort the data in the
+// key-value store for queries. Time series data is typically queryied by metric
+// name and a timespan; we therefore expose the metric name and the timestamp
+// prominently on the key.
+//
+// The precise formula for a time series key is:
+//   [system key prefix]tsd[metric name][resolution][timestamp][source key]
+//
+// The metric name is an arbitrary string identifying the metric, although the
+// ts system may enforce naming rules at a higher level. This string is binary
+// encoded in the key.
+//
+// The resolution refers to the sample duration at which data is stored. A
+// perfect resolution sample would keep the original timestamp and value for
+// every data point; lower resolution samples aggregate data points which fall
+// within the same sample period, keeping only the sum and count of all data
+// points. Multiple resolutions may be kept for a single metric; lower
+// resolutions are used to quickly query larger timespans, while higher
+// resolutions are used to provide more detail for smaller timespans. The
+// resolution is denoted in the key by a three character string.
+//
+// The timestamp is the first timestamp for which this key might contain data.
+// Multiple data samples are stored at the same key; a typical configuration
+// might have a ten second sample period, with all samples for the same hour
+// stored in the same key.
+//
+// The source is a possibly empty string which identifies the source for a
+// metric.  For example, a metric may describe an overall statistic for a
+// cluster of machines, but the data from each individual machine is stored
+// separately using a different source name. The source key is appended to the
+// end of the key; as such, keys for different sources with the same
+// metric name and timestamp will sort adjacently in the global key value store.
+// In this way, a metric can be quickly queried across all sources.
+var (
+	// KeyDataPrefix is the key prefix for time series data keys.
+	KeyDataPrefix = proto.MakeKey(engine.KeySystemPrefix, proto.Key("tsd"))
+	// KeyResolutionLength is the required length of resolution key components.
+	KeyResolutionLength = 3
+	// KeyResolutionTenSeconds is the resolution key component for keys which
+	// store data with a ten second sample duration.
+	KeyResolutionTenSeconds = proto.Key("10s")
```

Comment:
Don't export it until you need to use it outside of this package. We currently export way to much stuff unnecessarily.